src/next.md

```diff
@@ -7,6 +7,185 @@
+~~~
+
+### Exclude jtreg tests using `@ignore`
+
+The `@ignore` keyword is used in the test source code. This is mainly used for tests that are so broken that they may be harmful or useless. Examples can be tests that don't compile because something changed in the platform; or a test which might remove your `/etc/shadow`. Use `@ignore` with a bug reference in the test case to prevent the test from being run.
+
+~~~java
+/**
+ *  @test
+ *  @ignore 4711
+ *
+~~~
+
+In this example, `MyTest.java` is excluded, tracked by bug `JDK-4711`. `@ignore` should always be placed directly before the first `@run` line in the test.
+
+### Dealing with JBS bugs for test exclusion
+
+ProblemListing and `@ignore`-ing are done in the JDK source tree, that means a check-in into the repository is needed. This in turn means that a unique JBS issue and a code review are needed. This is a good thing since it makes test problems visible.
+
+* **Code review**: This is considered a trivial change. It only needs a review from one official Reviewer and you don't need to wait 24 hours before commit, even in areas that usually have stricter rules around reviews.
+* **JBS issue**: A JBS issue is obviously created for the bug that caused the failure, we call this the _main issue_. To exclude the test, create a subtask to the main issue. Also add the label `problemlist` to the main issue.
+
+The fix for the main issue should remove the test from the ProblemList or remove the `@ignore` keyword from the test.
+
+#### Triage excluded issues
+
+After a failure is handled by excluding a test, the main JBS issue should be re-triaged and possibly given a new priority. This should be handled by the standard triage process. A test exclusion results in an outage in our testing. This outage should be taken into consideration when triaging, in addition to the impact of the bug itself.
+
+## Backing Out a Change
+
+If a change cause a regression that can't be fixed within reasonable time the best way to handle the regression can be to back out the change. Backing out means that the inverse (anti-delta) of the change is pushed to effectively undo the change in the repository. There are two parts to this task, how to do the bookkeeping in JBS, and how to do the actual backout in mercurial.
+
+The backout is a regular change and will have to go through the standard code review process, but is considered a trivial change and thus it requires only one Reviewer and will avoid the 24h code review window, even for areas where stricter rules usually applies. The rationale is that a backout is usually urgent in nature and the change itself is automatically created by hg, and reviewed by the person who is performing the backout, so even though only one additional Reviewer is required the change will in practice get two reviews.
+
+### How to work with JBS when a change is backed out
+
+#. Close the original JBS issue **(O)**.
+   * "Verify" the issue and choose "Fix Failed".
```

Comment:
I have strong objections to fix failed ever being used and oppose it being recommended here. Unless the fixer and their reviewers completely failed at their job what you usually have is some other problem caused by the fix and the fix actually succeeded.

Reply:
I guess this is a question for those who normally handle fix verification and may have scripts that look for different verifications.

There's only four values to choose from: "None", "Verified", "Not verified", and "Fix failed".
"Verified" means that the fix solved the problem and no more action is required, so this is clearly not right. "Not verified" seems wrong since it actually was verified that the fix caused problems - or it wouldn't need to be backed out. "None" could be used in my mind, but I can imagine that there are filters that treats "None" as issues that needs verification. So changing to using this would probably cause problems. That leaves "Fix failed".

Maybe Joe knows why this was designed as it is?  Anyhow, it is the current process and we need to bring it up with the right people before changing it.

Reply:
AFAIK, The verification process isn't part of any OpenJDK process and is used/done mainly internally by Oracle. why do you think that verification status should be set for all backed out issues?

Reply:
The text is based on an ancient definition of how to back out changes. That text was JDK wide (not only HotSpot) but yes, internal to Oracle. If verification is not important or done for all issues I'd be happy to remove that sentence.

Reply:
There has been a longer discussion in the email thread on guide-dev that for some reason hasn't been copied here. From that and other discussions I now have a directive on this issue; The verification field is both useful and used, also for non-Oracle issues. It's not an Oracle internal field and anyone in the community can use it - it's actually documented on the OpenJDK wiki even though it's not always easy to find information in a wiki. For this reason it's actually even more important to clarify how it's intended to be used in the cases where we use it. Therefore we should include the verification part in this description as it is part of the process.

As for using "Fix failed"; this was very well explained by Igor in the same mail thread: this is the value that means that the bug still exists - and it does since the fix was backed out. It doesn't mean that the fix itself was bad or broken.

Reply:
just for the record, the email thread @JesperIRL referred to starts [here](https://mail.openjdk.java.net/pipermail/guide-dev/2020-July/000192.html).